bubble_sort.s

```diff
@@ -0,0 +1,138 @@
+	# printf insertarr
+        la      $t0, insertarr		# load address of insertarr into $t0
+        lw      $a0, 0($t0)             # load data from address in $t0 into $a0
+        li      $v0, 4                  # code of print_string
+        syscall  
+
+	# scanf num elements of array
+	li 	$t0 , 0			# t0 = 1 - counter
+	lw 	$t3 , num		# t3 = num
+	scanarr:			# beginig of for
+		li 	$v0 , 5			# code of read_int
+		syscall				# call scanf for int
+		sw 	$v0 , ($s0) 		# store int to the t1
+		add 	$s0 , $s0 , 4		# t1 += sizeof(int)
+		add 	$t0 , $t0 , 1		# t0++
+	blt 	$t0 , $t3 , scanarr	# condtiton of for
+
+	# now we have number of elements in num and address of array in s1
+	# calling bubble_sort
+	addi	$sp , $sp , -4		# move stack pointer on 1 word
+	sw	$ra , ($sp) 		# saving ra
+	lw	$a1 , num		# a1 = num
+	move 	$a0 , $s1		# a0 = address of array
+	jal bubble_sort
+
+	# printing array after sort
+	# printing main resbody
+        la      $t0, resbody            # load address of resbody into $t0
+        lw      $a0, 0($t0)             # load data from address in $t0 to $a0
+        li      $v0, 4                  # code of print_string
+        syscall
+
+	#print res
+	li      $t0 , 0                 # t0 = 1 - counter
+        lw      $t3 , num               # t3 = num
+        printarr:                        # beginig of for
+                li      $v0 , 4                 # code of print_string
+		la	$t7 , res		# load address of res in $t7
+		lw	$a0 , ($t7)		# load data from address in $t7 to $a0
+		syscall				# print " "
+		li	$v0 , 1
+		lw	$a0 , ($s1)		# a1 = *t1 - argument of print
+                syscall                         # call print for int
+                add     $s1 , $s1 , 4           # t1 += sizeof(int)
+                add     $t0 , $t0 , 1           # t0++
+        blt     $t0 , $t3 , printarr     # condtiton of for
+
+	# exit
+	addi	$sp , $sp , 4		# sp += sizeof(int)
+	lw	$ra , ($sp)		# restore ar
+	li 	$v0 , 10		# code of exit
+	syscall
+
+
+.text
+bubble_sort:
+	# for(i = 0 , i < num - 1 , i++)
+	#	for(j = 0 , j < num - 1 , j++)
+	#		if(a[j] > a[j + 1])
+	#			swap(a[j] , a[j + 1])
+	
+        # There are array address in a0 and number of elements in a1
+
+	move	$t2 , $a0		# t2 = a0 - it is a current array pointer
+	li      $t0 , 1                 # t0 = 1 - first counter
+	forsort1:			# begining of first for
+		li      $t1 , 1                 # t1 = 1 - second counter 
+		move	$t2 , $a0		# t2 = a0
+		forsort2:			# begining of second for		
+			lw	$t3 , ($t2)		# t3 = *t2
+			add	$t2 , $t2 , 4		# t2 += sizeof(int)
+			lw	$t4 , ($t2)		# t4 = *t2
+			ble	$t3 , $t4 , dontswap	# if t3 <= t4 don't swap
+			# swap section
+			sw	$t3 , ($t2)		# *t2 = t3
+			sw	$t4 , -4($t2)		# *(t2 - 1) = t4
+			# end of swap section
+			dontswap:	
+			add $t1 , $t1 , 1		# t1++ - counter 
+		blt	$t1 , $a1 , forsort2	# condition of second for
+		add 	$t0 , $t0 , 1		# t0++ - counter
+	blt	$t0 , $a1 , forsort1	# condition of first for
+
+
```

Comment:
I'd put a simple test here in terms whether array is sorted or not.
If array is sorted, we do `jr $ra` and halt simulation; otherwise, we print a error message and call a `sw $zero, ($zero)` (it throws an exception here).